src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/MonitorConnectionContext.java

```diff
@@ -111,11 +131,13 @@ void setConnectionValid(boolean connectionValid) {
           this.getFailureCount()));
     } else {
       this.setFailureCount(0);
```

Comment:
NIT: worth calling it resetFailureCount() ? so it has similar logic to the call below?